src/daemon/daemon.cpp

```diff
@@ -364,24 +366,40 @@ auto try_mem_size(const std::string& val) -> mp::optional<mp::MemorySize>
 }
 
 std::vector<mp::NetworkInterface> validate_extra_interfaces(const mp::LaunchRequest* request,
+                                                            std::vector<mp::NetworkInterfaceInfo> factory_networks,
                                                             mp::LaunchError& option_errors)
 {
-    // The associated bool indicates whether the interface mode is 'auto', i.e., needs to be configured by cloud-init.
     std::vector<mp::NetworkInterface> interfaces;
 
     for (const auto& net : request->network_options())
     {
+        // Check that the id the user specified is valid.
+        auto pred = [net](const mp::NetworkInterfaceInfo& info) { return info.id == net.id(); };
+        const auto& result = std::find_if(factory_networks.cbegin(), factory_networks.cend(), pred);
+
+        if (result == factory_networks.cend())
+        {
+            mpl::log(mpl::Level::debug, category, fmt::format("Invalid network id \"{}\"", net.id()));
```

Comment:
I would raise this to warning, so that the user gets it back with a single `-v`. Ideally they shouldn't need to increase verbosity to see the reason for the error, so I think it would even be ok to issue an error. The user should be getting this back without increasing verbosity. But the way it shows is not the cleanest:

```console
$ multipass launch --network asdf
[2020-12-07T13:07:03.321] [error] [daemon] Invalid network id "asdf"

launch failed: Invalid arguments supplied
Invalid network options supplied
```
This may feel strange without a `-v`.

Reply:
In order to improve the error message, we can do two things:
1. throw a `runtime_error` (as we do later in `create_vm`);
2. improve logging in the RPC, by adding a message to `LaunchError`.
What do you think?

Reply:
Wait, we can improve logging without changing the RPC, here: https://github.com/canonical/multipass/blob/validate-network-names/src/daemon/daemon.cpp#L2078

It's a matter of parsing the errors there and improve reporting. Does it make sense to you?